metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -264,24 +266,32 @@ def _get_semantic_model_for_measure(self, measure_reference: MeasureReference) -
         return semantic_models_where_measure_was_found[0]
 
     def get_joinable_metrics_for_semantic_model(
-        self, semantic_model: SemanticModel, using_join_path: Optional[SemanticModelJoinPath] = None
+        self, semantic_model: SemanticModel, using_join_path: SemanticModelJoinPath
     ) -> LinkableElementSet:
-        """Get the set of linkable metrics that can be joined to this semantic model."""
+        """Get the set of linkable metrics that can be joined to this semantic model.
+
+        From @courtneyholcomb re: `using_join_path`:
+            That would be the join path to get from the outer query to the metric subquery. If you query metric1
+            filtered by metric2 (grouped by entity1), it would be the join path to get from metric1 to entity1. If
+            entity1 is local to the semantic model where metric1's input measures are defined, no join path is
+            necessary.
+        """
         properties = frozenset({LinkableElementProperty.METRIC, LinkableElementProperty.JOINED})
-        if using_join_path:
+
+        join_path_has_path_links = len(using_join_path.path_elements) > 0
+        if join_path_has_path_links:
             assert semantic_model.reference == using_join_path.last_semantic_model_reference, (
                 "Last join path element should match semantic model when building LinkableMetrics. "
                 f"Got semantic model: {semantic_model.reference.semantic_model_name}; "
                 f"last join path element: {using_join_path.last_semantic_model_reference.semantic_model_name}",
             )
-            properties = properties.union(frozenset({LinkableElementProperty.MULTI_HOP}))
```

Comment:
We do want to keep this in the case when `join_path_has_path_links=True`. Currently you won't see this property in any of the snapshots because multi-hop options are temporarily disabled until we can test them more thoroughly. But we'll need this back when they're enabled again.

Reply:
Moved to a comment if you need it as a reminder to get rid of the dead code flag.